src/sentry/stacktraces/processing.py

```diff
@@ -325,10 +341,24 @@ def normalize_stacktraces_for_grouping(data, grouping_config=None) -> None:
                     frames, platform, exception_data
                 )
 
-    # normalize in-app
+    # normalize `in_app` values, noting whether event has a mix of in-app and system frames
     with sentry_sdk.start_span(op=op, description="normalize_in_app_stacktraces"):
+        frame_mixes = {"mixed": 0, "in-app-only": 0, "system-only": 0}
+
         for frames in stacktrace_frames:
-            _normalize_in_app(frames)
+            stacktrace_frame_mix = _normalize_in_app(frames)
+            frame_mixes[stacktrace_frame_mix] += 1
+
+    # store frame mix data on event for use in metrics
```

Comment:
I would move this block into it's own helper function and place it on line 351 after the for loop.
This is a preference.